SQL/0000-00-03-ConfigTables.sql

```diff
@@ -190,6 +192,8 @@ INSERT INTO Config (ConfigID, Value) SELECT ID, "/data/uploads/" FROM ConfigSett
 INSERT INTO Config (ConfigID, Value) SELECT ID, "/data/publication_uploads/" FROM ConfigSettings WHERE Name="publication_uploads";
 INSERT INTO Config (ConfigID, Value) SELECT ID, "/data/publication_uploads/to_be_deleted/" FROM ConfigSettings WHERE Name="publication_deletions";
 INSERT INTO Config (ConfigID, Value) SELECT ID, "%MINCToolsPath%" FROM ConfigSettings WHERE Name="MINCToolsPath";
+INSERT INTO Config (ConfigID, Value) SELECT ID, "/data/document_repository_uploads/" FROM ConfigSettings WHERE Name="documentRepositoryPath";
+INSERT INTO Config (ConfigID, Value) SELECT ID, "/data/data_release_uploads/" FROM ConfigSettings WHERE Name="dataReleasePath";
```

Comment:
Why don't we leave the default empty and update the code to have reasonable defaults if the value is null?

Reply:
Do you have an idea about what those constants would be? Where would be a reasonable place on a linux server for these files to go? Maybe somewhere under `/srv/`?

Reply:
I don't know what constants you're talking about.. I'm suggesting leaving them null.

Reply:
@driusan but you suggested to update the code to have reasonable defaults. what do you mean by that ?

Reply:
"defaults" was the wrong word.. I meant it should behave reasonably if the value is unset (ie. provide an error message pointing to the config setting if it's required or disable the feature if it's an optional thing similarly to how to the instrument_manager disables uploading if it doesn't have write access.)

Reply:
i see yeah that makes sens
should  I do that in this PR or... we wait for the centralised uploader ?

not sure

Reply:
What's the affect on this PR if the type is changed to `web_path` and the defaults are invalid paths?

Reply:
I have no idea, but for existing projects my patch will maintain the current directories until the project people switch it to another directory voluntarily.

My guess is when someone tries uploading something they will get a 500

Reply:
`web_path` only comes into affect when processing a POST Request to the config module. If the file path isn't set in the uploader, the module code will hopefully throw an error but these modules each do it in their own ad-hoc way.

The only bug that might occur is if someone tries to change a path in Config without changing the upload paths. At that point they'll need to correct every path before the submission succeeds.

Reply:
Is null a valid webpath ?

Reply:
https://github.com/aces/Loris/blob/4e3ee740fc4cb143f770ed4a4060377e327734ac/modules/configuration/ajax/process.php#L236-L242


So I don't think null works. In a meeting back when I was making this change we decided that an empty string will be used for default values. Discussion begins here: https://github.com/aces/Loris/pull/4023#issuecomment-464803516

Basically the calling code is responsible for doing something with the empty string when it finds it.

Reply:
There should never be an empty string in the config module. If it's empty, the row should be removed from the table. If there was a regression on the value being properly normalized it should be fixed.

Reply:
I'm sorry I wasn't being precise in my wording. You're right i don't think a blank entry will exist in the table so I'm incorrect about the defaults.

When a blank string is submitted for a field in the Config module, it has the effect of deleting that setting row from the table. 

So the behaviour in the code is like this:

If value is valid path --> write new entry to table
If value is empty string --> delete entry from table
If value is not empty and not valid --> reject input because invalid web path supplied.

Reply:
that's the expected behaviour, which means that it should go through the 'delete from table' logic and not the validPath code that you quoted above. It's not a "valid path" but the config module shouldn't reject it either.

Reply:
Yeah sorry my mind is still waking up. 😅 NULL should probably work as a default value.

Reply:
okai so if i understand correctly;
 - in RB set the paths to "ideal" values `/data/...`
 - in 0000 conf files do not add an entry into the Config table and let projects set a validpath
 - in the patch for existing projects, maintain status quo and discuss in the release the recommendation of moving the directories.


agreed ?

Reply:
Yes, though we'll need to provide some detail about also moving the files from one destination to another, not just changing the path.

Reply:
I don't know what the status quo is so I can't comment on the last point, but that sounds right.

Reply:
e.g. if there's `modules/document_repository/user_uploads/admin/meme.jpg` then the `admin/` folder will need to be moved to the new location or else it will be inaccessible.

Reply:
yes for sure but that would not be an issue in the current state because existing projects will have their path point to the old path.

it should only be a recommendation on our end